lib/net/pool.js

```diff
@@ -1883,6 +1898,119 @@ class Pool extends EventEmitter {
     }
   }
 
+  /**
+   * Handle peer `getcfilters` packet.
+   * @method
+   * @private
+   * @param {Peer} peer
+   * @param {GetCFiltersPacket} packet
+   */
+
+  async handleGetCFilters(peer, packet) {
+    if (!this.options.bip157)
+      return;
+
+    const stopHeight = await this.chain.getHeight(packet.stopHash);
+    // The stop hash doesn't exist
+    if (!stopHeight)
+      return;
+
+    if (stopHeight - packet.startHeight + 1 >= common.MAX_CFILTERS)
+      return;
+
+    const indexer = this.getFilterIndexer(filtersByVal[packet.filterType]);
+    if (!indexer)
+      return;
+
+    for (let i = packet.startHeight; i <= stopHeight; i++) {
+      const blockHash = await this.chain.getHash(i);
+      const filter = await indexer.getFilter(blockHash);
+      peer.sendCFilter(packet.filterType, blockHash, filter.filter);
+    }
+  }
+
+  /**
+   * Handle peer `getcfheaders` packet.
+   * @method
+   * @private
+   * @param {Peer} peer
+   * @param {GetCFHeadersPacket} packet
+   */
+
+  async handleGetCFHeaders(peer, packet) {
+    if (!this.options.bip157)
+      return;
+
+    const stopHeight = await this.chain.getHeight(packet.stopHash);
+    // The stop hash doesn't exist
+    if (stopHeight < 0)
+      return;
+
+    if (stopHeight - packet.startHeight >= common.MAX_CFHEADERS)
+      return;
+
+    const indexer = this.getFilterIndexer(filtersByVal[packet.filterType]);
+
+    if (!indexer)
+      return;
+
+    const startHash = await this.chain.getHash(packet.startHeight - 1);
+
+    let previousFilterHeader = consensus.ZERO_HASH;
+
+    if (startHash)
+      previousFilterHeader = await indexer.getFilterHeader(startHash);
+
+    const filterHashes = [];
+    for (let i = packet.startHeight; i <= stopHeight; i++) {
+      const blockHash = await this.chain.getHash(i);
+      const filterHash = await indexer.getFilterHash(blockHash);
+      if (!filterHash) {
+        return;
+      }
+      filterHashes.push(filterHash);
+    }
+
+    peer.sendCFHeaders(
+      packet.filterType,
+      packet.stopHash,
+      previousFilterHeader,
+      filterHashes);
+  }
+
+  /**
+   * Handle peer `getcfcheckpt` packet.
+   * @method
+   * @private
+   * @param {Peer} peer
+   * @param {GetCFCheckptPacket} packet
+   */
+
+  async handleGetCFCheckpt(peer, packet) {
+    if (!this.options.bip157)
+      return;
+
+    const stopHeight = await this.chain.getHeight(packet.stopHash);
+    // The stop hash doesn't exist
+    if (stopHeight == null)
+      return;
+
+    const filterHeaders = [];
+    const indexer = this.getFilterIndexer(filtersByVal[packet.filterType]);
+
+    if (!indexer)
+      return;
+
+    for (let height = 1000; height <= stopHeight; height += 1000) {
+      const blockHash = await this.chain.getHash(height);
+      if (!blockHash)
+        break;
+      const filterHeader = await indexer.getFilterHeader(blockHash);
+        filterHeaders.push(filterHeader);
```

Comment:
sorry i missed this before, spacing nit
```suggestion
      filterHeaders.push(filterHeader);
```